### ISSUE TITLE
Custom queries/filters in Reports

### ISSUE THREAD

**Issue by olemp** [Puzzlepart/did#799]
- [x] I have searched for existing issues to avoid creating duplicates

### TLDR
Custom saved queries (filters) in Reports

### Description
Users, especially management need a report filtered on the same filter parameters repeatedly - typically all employees in the same department.  
A feature for saving Report filters would be time saving and a great value-add for the Report page.

### Todos
- [x] Add smoke tests

#### Proposed UX
* Button to the right of the filter funnel, *Saved filters*
* List of saved filters (if any), plus a *Save current filter* button
![image](https://user-images.githubusercontent.com/7300548/107039565-5e642280-67be-11eb-9887-edfb8565b39e.png)
* Ability to give a custom name to the filter
![image](https://user-images.githubusercontent.com/7300548/107039647-763ba680-67be-11eb-899a-13b410e226bc.png)
* On save, the filter gets persisted to the current User object in the database, and becomes accessible wherever the user is logged in.
* *Delete Filter* feature (in progress)
* *Edit filter* feature not in scope for now

### Feature request source (Teams)
![image](https://user-images.githubusercontent.com/7606007/106145662-1a49a000-6176-11eb-8814-ae4c5bccc07a.png)
